core/src/main/java/org/springframework/security/oauth2/server/authorization/AuthorizationCodeOAuth2Authorization.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.oauth2.server.authorization;
+
+import org.springframework.util.Assert;
+
+import java.util.Map;
+
+/**
+ * An {@link OAuth2Authorization} implementation for authorization code grant. The authorization code must be provided
+ * in the {@code attributes} map.
+ *
+ * @author Krisztian Toth
+ */
+public class AuthorizationCodeOAuth2Authorization extends OAuth2Authorization {
```

Comment:
This class is not needed. The intended purpose for `OAuth2Authorization.attributes` is to store **non-common** attributes. The `code` parameter is `authorization_code` grant specific so it would be stored here using a well-known key.

Another example of what would be stored here is the `scope(s)` consented by the user during the authorization consent step #42.